taar/recommenders/legacy_recommender.py

```diff
@@ -0,0 +1,51 @@
+from ..recommenders import utils
+
+ADDON_LIST_BUCKET = 'telemetry-private-analysis-2'
+ADDON_LIST_KEY = 'taar/legacy/legacy_dict.json'
+
+
+class LegacyRecommender:
+    """ A recommender class that returns potential replacements for depricated legacy addons.
+
+    This will load a json file containing updated periodically for suggest web extension
+    alternatives to legacy addons not compatible with Fx 57.
+
+    This recommender may provide useful recommendations when collaborative_recommender
+    may not work.
+    """
+    def __init__(self):
+        self.legacy_replacements = utils.get_s3_json_content(ADDON_LIST_BUCKET,
+                                                             ADDON_LIST_KEY)
+
+    def can_recommend(self, client_data):
+        # We can't recommend if we don't have our data files.
+        if self.legacy_replacements is None:
+            return False
+
+        # get active addons from the client string and pull the legacy guids
+        client_addons = client_data.get('installed_addons', None)
+
+        # cant use this recommender is no addons are installed
+        if not client_addons:
+            return False
+
+        if not isinstance(client_addons, list):
+            return False
+
+        # if no member of the installed legacy addon guids can be found in the loaded resource,
+        # then no recommendation is possible
+        if not set.intersection(*[set(client_addons), set(self.legacy_replacements.keys())]):
+            return False
+
+        return True
+
+    def recommend(self, client_data, limit):
+        client_addons = client_data.get('installed_addons')
```

Comment:
Let's rename `client_addons` to `legacy_addons`  and use the  correct field here.